turbo/stages/bodydownload/body_algos.go

```diff
@@ -364,3 +381,55 @@ func (bd *BodyDownload) AddMinedBlock(block *types.Block) error {
 	bd.AddToPrefetch(block)
 	return nil
 }
+
+// GetHeader returns a header by either loading from the deliveriesH slice populated when running RequestMoreBodies
+// or if the code is continuing from a previous run and this isn't present, by reading from the DB as the RequestMoreBodies would have.
+// as the requestedLow count is incremented before a call to this function we need the process count so that we can anticipate this,
+// effectively reversing time a little to get the actual position we need in the slice prior to requestedLow being incremented
+func (bd *BodyDownload) GetHeader(blockNum uint64, blockReader services.FullBlockReader, tx kv.Tx) (*types.Header, common.Hash, error) {
+	var header *types.Header
+	if bd.deliveriesH[blockNum] != nil {
+		header = bd.deliveriesH[blockNum]
+	} else {
+		hash, err := rawdb.ReadCanonicalHash(tx, blockNum)
+		if err != nil {
+			return nil, common.Hash{}, err
+		}
+		header, err = blockReader.Header(context.Background(), tx, hash, blockNum)
+		if err != nil {
+			return nil, common.Hash{}, err
+		}
+		if header == nil {
+			return nil, common.Hash{}, fmt.Errorf("header not found: blockNum=%d, hash=%x, trace=%s", blockNum, hash, dbg.Stack())
+		}
+	}
+	return header, header.Hash(), nil
+}
+
+func (bd *BodyDownload) addBodyToBucket(tx kv.RwTx, key uint64, body *types.RawBody) error {
+	k := createBucketKey(key)
+	marshalled, err := json.Marshal(body)
```

Comment:
Don't think converting to JSON is really necessary, buckets do not have to have textual values, it can be any string, except an empty string

Reply:
Any suggestions for alternatives here?

Reply:
All in place now, found an issue with the encoding/decoding of `RawBody` which has been addressed now as well with a unit test wrapped around it.